src/main/java/domain/Race.java

```diff
@@ -2,42 +2,84 @@
-            RacingCar car = new RacingCar();
+    public static void createCars(RacingCarNameRequest nameList) {
+
+        for (String name : nameList.getNameList()) {
+            isNameValid(name);
+            RacingCar car = new RacingCar(name);
             racingCarList.add(car);
         }
     }
 
     public static void race() {
         for (RacingCar racingCar : racingCarList) {
-            boolean coin = CarNavigator.flipCoin(CarNavigator.getRandomNumber());
-            racingCar.moveOrStop(coin);
+            MovingStrategy movingStrategy = new RandomMovingStrategy();
+            racingCar.moveOrStop(movingStrategy);
         }
     }
 
     public static String[] getEachResult() {
         String[] result = new String[racingCarList.size()];
         for (int i = 0; i < racingCarList.size(); i++) {
-            result[i] = "-".repeat(racingCarList.get(i).getDistance());
+            result[i] = racingCarList.get(i).getResult();
         }
         return result;
     }
-
-    public static List<String[]> playRace(int[] input) {
-        int carCount = input[0];
-        int round = input[1];
+    
+    public static GameResultResponse playRace(RacingRoundRequest roundRequest) {
+        int round = roundRequest.getNumberOfRounds();
         List<String[]> totalResult = new ArrayList<>();
-        Race.createCar(carCount);
         for (int i = 0; i < round; i++) {
-            Race.race();
+            race();
             totalResult.add(getEachResult());
         }
-        return totalResult;
+        return new GameResultResponse(totalResult);
+    }
+
+    private static void isNameValid(String name) {
+        if (!RacingCar.isNameValid(name)){
+            throw new IllegalArgumentException("자동차 이름은 5자를 초과할 수 없습니다.");
+        }
+    }
+
+    public static List<String> getWinnerList() {
+        List<String> winnerList = new ArrayList<>();
+        int[] matchResult = getMatchResult();
+        int maxDistance = getMaxDistance(matchResult);
+        for (RacingCar racingCar : racingCarList) {
+            checkWinner(winnerList, maxDistance, racingCar);
+        }
+        return winnerList;
+    }
+
+    private static void checkWinner(List<String> winnerList, int maxDistance, RacingCar racingCar) {
+        if (Objects.equals(maxDistance, racingCar.getDistance())) {
+            winnerList.add(racingCar.getName());
+        }
+    }
+
+    public static int getMaxDistance(int[] matchResult) {
+
+        int maxDistance = matchResult[0];
+        for (int distance : matchResult) {
+            if (distance > maxDistance) {
+                maxDistance = distance;
+            }
+        }
+        return maxDistance;
+    }
+
+    private static int[] getMatchResult() {
+        int[] matchResult = new int[racingCarList.size()];
+        for (int i = 0; i < racingCarList.size(); i++) {
+            matchResult[i] = racingCarList.get(i).getDistance();
+        }
+        return matchResult;
     }
```

Comment:
Stream을 활용한다면 아래와 같이도 정의해볼수 있을것 같네요 😄 
```suggestion
    racingCarList.stream().map(::getDistance).mapToint().max()...
```

Reply:
@sah3122 피드백 감사합니다 :) 다만, 이번 미션 수행하면서 포비가 스트림을 최대한 사용하지 않고 해보는 게 어떻겠냐는 조언이 있어서요, 해당 사항은 반영하지 않아도 괜찮을까요?

Reply:
해당 조언을 제가 몰랐네요 ㅎㅎ 
현재처럼 구현해주셔도 됩니다 !